app/services/move_to_problem_queue.rb

```diff
@@ -12,6 +12,10 @@ def initialize(order_detail, force: false)
   end
 
   def move!
+    # Some scopes may accidentally try send already-complete orders to the queue.
+    # This protects against sending duplicate emails to things already in the queue.
+    return unless OrderDetail.pending.include?(@order_detail)
```

Comment:
Rather than querying a scope on the class when we have the object handy, can we introduce `OrderDetail#pending?` here, along the lines of:

```
def pending?
  order.purchased? && state.in?(["new", "inprocess"])
end
```

That way we can call `return unless @order_detail.pending?` which seems more natural to me.

Reply:
Yeah, definitely more natural. I think I was a little annoyed at how much duplication there was in these scope, that adding a predicate seemed duplicative as well. I've now gotten rid of the `pending` method, which is identical to `new_or_inprocess`.